docs/HowToDevelopLoadableApps.md

```diff
@@ -0,0 +1,216 @@
+
+ELF is stored in flash partition and loaded into ram at runtime and hence we need to reserve
+both RAM & FLASH memories.
+
+#### Flash Partition Settings
+
+- Add a new flash partition for each application using the 3 configurations given below.
+	1. The 'size' should be large enough to accomodate the binary size (CONFIG_FLASH_PART_SIZE).
+	2. The 'type' should be of the type "bin" (CONFIG_FLASH_PART_TYPE).
+	3. The 'name' can be the name of the application (CONFIG_FLASH_PART_NAME).
+
+  In below scenario, the apps namely micom & wifi of sizes 1MB each are added. The two instances (wifi,wifi) are meant for FOTA update of binaries (if supported). One will contain the actual binary, whereas the other, the updated binary.
+
+```
+CONFIG_FLASH_PART_SIZE="2048,512,1024,1024,1024,1024,1024,128,128,"
+CONFIG_FLASH_PART_TYPE="kernel,none,bin,bin,bin,bin,smartfs,ftl,config,"
+CONFIG_FLASH_PART_NAME="kernel,app,micom,micom,wifi,wifi,userfs,ftl,config,"
+```
+
+#### RAM Memory Settings
+
+- Text & data sections of loadable apps will be allocated from user RAM region.
+- So find out the memory region large enough to load the appliacations.
+- Add its address & size to CONFIG_RAM_REGIONx_START & CONFIG_RAM_REGIONx_SIZE (if not already present).
+
+  In below scenario, we added a region which starts at 0x80200000 of size 30MB.
+
+```
+CONFIG_RAM_REGIONx_START="0x20200000,0x80200000"
+CONFIG_RAM_REGIONx_SIZE="262144,31457280"
+```
+
+## Steps to add new loadable apps
+
+### Configuration Settings
+
+- Follow the requirements as stated here ([Configuration Settings](#configuration-settings)) to enable all configurations related to loadable apps.
+
+### Memory Settings
+
+- Follow the requirements as stated here ([Memory Requirements](#memory-requirements)) for reserving flash & ram partitions for adding new applications.
+
+### Code Changes Required
+
+- Add required apps to "loadable_apps/loadable_sample/<app_name>"
+
+#### Makefile
+
+- Make sure to generate elf type binary for each newly added application.
+  In <os/loadable_apps/loadable_sample/app/Makefile>
+
+```
+ifeq ($(CONFIG_APP_BINARY_SEPARATION),y)
+
+BIN = wifi
+BIN_TYPE = ELF
+BIN_VER = 20190412
+DYNAMIC_RAM_SIZE = 512000
+KERNEL_VER = 2.0
+STACKSIZE = 8192
+PRIORITY = 180
+
+include $(TOPDIR)/$(LOADABLEDIR)/loadable.mk
+endif # CONFIG_APP_BINARY_SEPARATION
+
+```
+Where,
+- BIN_TYPE: Binary Type (should be elf)
+- STACKSIZE: The first task will have a stack of this size
+- PRIORITY: The first (main) task of the app will be created with this priority
+- DYNAMIC_RAM_SIZE: This is the total amount of ram that will be available to the app. The developer must assess the max ram requirement for the app. During run time, if the memory requirement exceeds this value, then the app will fail. In that case the user might need to provide a higher value for this and rebuild the app and run it again.
+
+These values will be used by the kernel to start the first process of the application.
```

Comment:
How about chaning `kernel` to `binary manager`?

Reply:
yes i think that would be better